File: python/p002.py
#!/usr/bin/env python3

"""Problem 2: Even Fibonacci Numbers"""


def main():
    limit = 4_000_000
    curr_even, prev_even, Σ = 2, 0, 0

    while curr_even <= limit:
        Σ += curr_even
        prev_even, curr_even = curr_even, 4 * curr_even + prev_even
    return Σ


if __name__ == "__main__":
    print(main())

File: go/p002.go
package main

import "fmt"

func main() {
	const limit = 4_000_000

	prevEven, currEven, sum := 0, 2, 0

	for currEven <= limit {
		sum += currEven
		prevEven, currEven = currEven, 4*currEven+prevEven
	}

	fmt.Println(sum)
}
